sympy/polys/matrices/sdm.py

```diff
@@ -31,6 +31,34 @@ def __init__(self, elemsdict, shape, domain):
         if not all(0 <= c < n for row in self.values() for c in row):
             raise DDMBadInputError("Column out of range")
 
+    def getitem(self, i, j):
+        try:
+            return self[i][j]
+        except KeyError:
+            m, n = self.shape
+            if -m <= i < m and -n <= j < n:
+                try:
+                    return self[i % m][j % n]
+                except KeyError:
+                    return self.domain.zero
+            else:
+                raise IndexError("index out of range")
+
+    def extract_slice(self, slice1, slice2):
+        m, n = self.shape
+        ri = range(m)[slice1]
+        ci = range(n)[slice2]
+
+        sdm = {}
+        for i, row in self.items():
+            if row:
+                if i in ri:
+                    row = {ci.index(j): e for j, e in row.items() if j in ci}
+                    if row:
+                        sdm[ri.index(i)] = row
```

Comment:
The first `if row` check is not needed. The row will never be empty in a valid `SDM`. The idea is that if we check for zero rows when creating a `SDM` we never need to check it afterwards.

Reply:
> The first `if row` check is not needed. The row will never be empty in a valid `SDM`. 
>
Sure, will remove the `if` condition.
Thanks. 

> The idea is that if we check for zero rows when creating a `SDM` we never need to check it afterwards.
>
Indeed, this looks very logical.